source/scripts/app.js

```diff
@@ -751,6 +781,77 @@ function connectRecipeAction() {
   });
 }
 
+/**
+ * Adds an event listener to the "Save Changes" button in the "Edit Cookbook"
+ * page.
+ * @function buttonsEditCookbook
+ */
+function buttonsEditCookbook() {
+  // Get the "Save Changes" button
+  "use strict";
+  let templatePage = document.querySelector("edit-cookbook");
+  let shadow = templatePage.shadowRoot;
+  let saveButton = shadow
+    .querySelector("div")
+    .children[3].getElementsByTagName("button")[0];
+
+  let cancelButton = shadow
+    .querySelector("div")
+    .children[2].getElementsByTagName("button")[0];
+
+  saveButton.addEventListener("click", async () => {
+    // Get the Title and the Description
+    let templatePage = document.querySelector("edit-cookbook");
+    let shadow = templatePage.shadowRoot;
+    let mainDiv = shadow.querySelector("div.input-container");
+
+    // Gets the div by index (first div = 0, second div = 1)
+    let title = mainDiv.children[0].getElementsByTagName("input")[0].value;
+    let description =
+      mainDiv.children[1].getElementsByTagName("input")[0].value;
+
+    // Nothing to update
+    if (title === NO_INPUT && description === NO_INPUT) {
+      router.navigate("cook-book");
+    }
+    // Only update description
+    else if (title === NO_INPUT && description !== NO_INPUT) {
+      // Place into storage
+      await indexedDb.editCookbook(
+        COOKBOOK_TO_EDIT,
+        COOKBOOK_TO_EDIT,
+        description
+      );
+
+      // Update the cookbooks
+      await populateCookbooksPage();
+
+      // Set the textbox fields to original format
+      mainDiv.children[0].getElementsByTagName("input")[0].value = NO_INPUT;
+      mainDiv.children[1].getElementsByTagName("input")[0].value = NO_INPUT;
+
+      // Go back to cookbook page
+      router.navigate("cook-book");
+    } else {
+      await indexedDb.editCookbook(COOKBOOK_TO_EDIT, title, description);
+
+      await populateCookbooksPage();
+
+      mainDiv.children[0].getElementsByTagName("input")[0].value = NO_INPUT;
+      mainDiv.children[1].getElementsByTagName("input")[0].value = NO_INPUT;
+
+      router.navigate("cook-book");
+    }
+
+    cancelButton.addEventListener("click", () => {
+      mainDiv.children[0].getElementsByTagName("input")[0].value = NO_INPUT;
+      mainDiv.children[1].getElementsByTagName("input")[0].value = NO_INPUT;
+
+      router.navigate("cook-book");
+    });
+  });
```

Comment:
I found the bug that made your cancel button not work
```suggestion
  });

  cancelButton.addEventListener("click", () => {
    let mainDiv = shadow.querySelector("div.input-container");
    mainDiv.children[0].getElementsByTagName("input")[0].value = NO_INPUT;
    mainDiv.children[1].getElementsByTagName("input")[0].value = NO_INPUT;

    router.navigate("cook-book");
  });
```